lms/djangoapps/ccx/views.py

```diff
@@ -84,14 +85,19 @@ def wrapper(request, course_id):
         ccx = None
         if isinstance(course_key, CCXLocator):
             ccx_id = course_key.ccx
-            ccx = CustomCourseForEdX.objects.get(pk=ccx_id)
-            course_key = ccx.course_id
+            try:
+                ccx = CustomCourseForEdX.objects.get(pk=ccx_id)
+                course_key = ccx.course_id
+            except CustomCourseForEdX.DoesNotExist:
+                raise Http404
 
         course = get_course_by_id(course_key, depth=None)
         is_staff = has_access(request.user, 'staff', course)
         is_instructor = has_access(request.user, 'instructor', course)
 
-        if is_staff or is_instructor:
+        if not course.enable_ccx:
+            raise Http404
```

Comment:
when does this case happen? when a CCX has been created and then the CCX option disabled in advanced settings?